Adding a Larger Framework for the Player API

katiesweet opened the pull request:
I'M ASKING FOR CODE REVIEWS FROM:
Hearts Logic: @trevorjewkes
Spades Logic: @adamtew
8's Logic: @Shemarama
PlayerNetworking: @michaelkamerath

@Frangello and I added a few functions to the Player API, and added a Card.hpp as well as a PlayerAPI.cpp with a few functions filled in. This code should actually compile and be usable, unlike the initial commit. There isn't any tests for this code yet, but I still wanted to get it available to you all as soon as possible since you will be using it.

Also, per @C-Merrill request, we changed the Card class to create cards of values 2-14 instead of 1-13  Ace high in our games, so be aware of that.

Once we get 3/4 of you to approve, I will create another pull request with michaelkamerath/SharedServer.

michaelkamerath commented on PlayerAPI/Card.cpp:
```diff
@@ -1,96 +1,82 @@
+}
 
-class Card
+Card::Card(Suit su, int val)
```
Please change this to an initializer list rather than assigning in the body of the constructor

michaelkamerath commented on PlayerAPI/Card.cpp:
```diff
@@ -1,96 +1,82 @@
+    // Values are from 2 to 14 corrosponding from 2 - Ace.
+    std::cout << "Error: Tried to instantiate a card of undefined value"
+              << std::endl;
+    throw(2);
```
Please throw a std::exception rather than an int

katiesweet commented:
Thanks for the review @michaelkamerath. I've made the changes you've suggested, as well as:
1. Added a bunch of comments
2. Removed getters/setters for the ip address, as they shouldn't be necessary (The ip address is set in the Player constructor and shouldn't be change. It also should only need to be accessed by the networking functions inside the player class)
3. I removed the setBid() function because requestBid() already gets and sets the bid from the client.
